src/v/iceberg/manifest_avro.cc

```diff
@@ -0,0 +1,180 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/manifest_avro.h"
+
+#include "base/units.h"
+#include "bytes/iobuf.h"
+#include "iceberg/avro_utils.h"
+#include "iceberg/datatypes_json.h"
+#include "iceberg/json_utils.h"
+#include "iceberg/manifest.h"
+#include "iceberg/schema.h"
+#include "iceberg/schema_json.h"
+#include "strings/string_switch.h"
+
+#include <avro/DataFile.hh>
+
+namespace iceberg {
+
+namespace {
+ss::sstring schema_to_json_str(const schema& s) {
+    json::StringBuffer buf;
+    json::Writer<json::StringBuffer> w(buf);
+    rjson_serialize(w, s);
+    return buf.GetString();
+}
+schema schema_from_str(const std::string& s) {
+    json::Document parsed_schema;
+    parsed_schema.Parse(s);
+    return parse_schema(parsed_schema);
+}
+
+ss::sstring content_type_to_str(manifest_content_type t) {
+    switch (t) {
+    case manifest_content_type::data:
+        return "data";
+    case manifest_content_type::deletes:
+        return "deletes";
+    }
+}
+manifest_content_type content_type_from_str(std::string_view s) {
+    return string_switch<manifest_content_type>(s)
+      .match("data", manifest_content_type::data)
+      .match("deletes", manifest_content_type::deletes);
+}
+
+ss::sstring format_to_str(format_version v) {
+    switch (v) {
+    case format_version::v1:
+        return "1";
+    case format_version::v2:
+        return "2";
+    }
+}
+format_version format_from_str(std::string_view s) {
+    return string_switch<format_version>(s)
+      .match("1", format_version::v1)
```

Comment:
nit: change `format_to_string` to return `constexpr std::string_view` and change this to be `.match(to_string(format_version::v1), format_version::v1)`